### PR TITLE
MONGOID-3326 Test to show that #after_initialize issue is resolved

### PR DESCRIPTION
**Pull request by estolfo**

### CODE REVIEW

File: spec/mongoid/interceptable_spec.rb [hunk 1]
```diff
@@ -170,6 +170,65 @@ class TestClass
...
+
+      before do
+        Book.destroy_all
+      end
+
+      after do
+        Book.destroy_all
+      end
+
+      let(:book) do
```
Comment (simi) [inline spec/mongoid/interceptable_spec.rb]: What about this indentation style :question:

```ruby
let(:book) do
  book = Book.new({
    :pages => [
      {
        content: "Page 1",
        notes: [
          { message: "Page 1 / Note A" },
          { message: "Page 1 / Note B" }
        ]
      },
      {
        content: "Page 2",
        notes: [
          { message: "Page 2 / Note A" },
          { message: "Page 2 / Note B" }
        ]
      }
    ]
  })
  book.id = '123'
  book.save
  book
end
```

File: spec/mongoid/interceptable_spec.rb [hunk 2]
```diff
@@ -170,6 +170,65 @@ class TestClass
     it "runs after document instantiation" do
       expect(game.name).to eq("Testing")
     end
+
+    context 'when the document is embedded' do
+
+      before do
+        Book.destroy_all
```
Comment (simi) [inline spec/mongoid/interceptable_spec.rb]: Why is this needed? AFAIK MongoDB is [purged](https://github.com/mongodb/mongoid/blob/095433fa53fc5f770fedb8d28b57fa33b2b2c4dc/spec/spec_helper.rb#L124-L127) before any test.

File: spec/mongoid/interceptable_spec.rb [hunk 3]
```diff
@@ -170,6 +170,65 @@ class TestClass
...
+                                             ]
+                                }
+                            ]
+                        })
+        book.id = '123'
+        book.save
+        book
+      end
+
+      let(:new_message) do
```
Comment (simi) [inline spec/mongoid/interceptable_spec.rb]: *just nitpick: You can save two lines by using inline block here.*

```ruby
let(:new_message) { 'Note C' }
```

File: spec/mongoid/interceptable_spec.rb (hunk 1 above)
Comment (durran) [inline spec/mongoid/interceptable_spec.rb]: Agreed - please use 2 character soft tabs please.

Comment (estolfo) [inline spec/mongoid/interceptable_spec.rb]: The spacing was leftover from copy-and-pasting from the ticket info. I've updated it.

File: spec/mongoid/interceptable_spec.rb (hunk 2 above)
Comment (estolfo) [inline spec/mongoid/interceptable_spec.rb]: Took that out.
